test/integration/integration_admin_test.cc

```diff
@@ -414,6 +420,42 @@ TEST_P(IntegrationAdminTest, Admin) {
   envoy::admin::v2alpha::SecretsConfigDump secret_config_dump;
   config_dump.configs(5).UnpackTo(&secret_config_dump);
   EXPECT_EQ("secret_static_0", secret_config_dump.static_secrets(0).name());
+
+  // Validate that the inboundonly does not drain the outbound listeners.
+  response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "POST",
+                                                "/drain_listeners?inboundonly", "",
+                                                downstreamProtocol(), version_);
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().Status()->value().getStringView());
+  EXPECT_EQ("text/plain; charset=UTF-8", ContentType(response));
+  EXPECT_EQ(0, test_server_->counter("listener_manager.listener_removed")->value());
+
+  // Validate that the outbound listener is drained on drain listeners.
+  response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "POST", "/drain_listeners", "",
+                                                downstreamProtocol(), version_);
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().Status()->value().getStringView());
+  EXPECT_EQ("text/plain; charset=UTF-8", ContentType(response));
+  test_server_->waitForCounterGe("listener_manager.listener_removed", 1);
```

Comment:
How long do these tests take to run with real time? I'm guessing many seconds because of draining? I think you will likely need to modify the drain time for the listener or Envoy to make this shorter?

More generally, can you potentially do a test with actual requests to make sure that the listener is actually draining and then gone when we do this admin command?

Reply:
It runs for 1s.  For integration test server it is set here https://github.com/envoyproxy/envoy/blob/master/test/integration/server.cc#L42.

My intention was not not test the actual draining behaviour - but ensure that it is triggered as draining api would have already been tested. Added a new test now. Please see if that is what you are expecting?